pinot-spi/src/main/java/org/apache/pinot/spi/config/table/TableConfig.java

```diff
@@ -89,6 +91,7 @@ public class TableConfig extends BaseJsonConfig {
   @JsonPropertyDescription(value = "Point to an existing instance partitions")
   private Map<InstancePartitionsType, String> _instancePartitionsMap;
 
+  private Map<InstancePartitionsType, SegmentAssignmentConfig> _segmentAssignmentConfigMap;
```

Comment:
Since the key is `InstancePartitionsType`, we won't be able to configure the assignment strategy for the tiers. Consider changing the key to `String` to be able to represent tiers

Reply:
updated